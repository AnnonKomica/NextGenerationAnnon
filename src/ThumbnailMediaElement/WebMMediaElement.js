// @flow
import React from "react";
import ThumbnailMediaElement from "./ThumbnailMediaElement";
import ClassNameUtil from "../utils/ClassNameUtil";

export default class WebMMediaElement extends ThumbnailMediaElement {
    _getRawMediaElement(): ?Object {
        const {
            showRaw,
            detailLoaded
        } = this.state;

        if (!showRaw) {
            return null;
        }

        const containerClass = ClassNameUtil.classNames(["thumbnail-media__raw-media-container"], {
            "hide": !detailLoaded,
        });


        return (
            <div className={containerClass}>
                <video
                    autoPlay
                    className="thumbnail-media__raw-media--webm"
                    controls
                    src={this.props.rawMediaSrc}
                    onCanPlay={this._handleRawElementLoad}
                />
                <div
                    className="thumbnail-media__raw-media-close-button"
                    onClick={this._handleRawElementClick}
                >
                    x
                </div>
            </div>
        );
    }
}
