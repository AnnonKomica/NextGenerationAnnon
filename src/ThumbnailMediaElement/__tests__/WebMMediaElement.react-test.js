import React from "react";
import renderer from "react-test-renderer";

import WebMMediaElement from "../WebMMediaElement";

// eslint-disable-next-line
test("WebMMediaElement DOM should render correctly", () => {
    const component = renderer.create(
        <WebMMediaElement
            thumbnailSrc="thumbnailSrc"
            rawMediaSrc="rawMediaSrc"
        />
    );

    let tree = component.toJSON();

    // eslint-disable-next-line
    expect(tree).toMatchSnapshot();

    tree.children[0].props.onClick();
    tree = component.toJSON();

    // eslint-disable-next-line
    expect(tree).toMatchSnapshot();

    tree.children[1].children[0].props.onCanPlay();
    tree = component.toJSON();

    // eslint-disable-next-line
    expect(tree).toMatchSnapshot();


    tree.children[1].children[1].props.onClick();
    tree = component.toJSON();

    // eslint-disable-next-line
    expect(tree).toMatchSnapshot();
});
