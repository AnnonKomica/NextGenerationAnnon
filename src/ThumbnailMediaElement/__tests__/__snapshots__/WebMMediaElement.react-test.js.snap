exports[`test WebMMediaElement DOM should render correctly 1`] = `
<div>
  <img
    className="thumbnail-media__thumbnail"
    onClick={[Function]}
    src="thumbnailSrc" />
</div>
`;

exports[`test WebMMediaElement DOM should render correctly 2`] = `
<div>
  <img
    className="thumbnail-media__thumbnail thumbnail-media__thumbnail--raw-loading"
    onClick={[Function]}
    src="thumbnailSrc" />
  <div
    className="thumbnail-media__raw-media-container thumbnail-media__raw-media--loading">
    <video
      autoPlay={true}
      className="thumbnail-media__raw-media--webm"
      controls={true}
      onCanPlay={[Function]}
      src="rawMediaSrc" />
    <div
      className="thumbnail-media__raw-media-close-button"
      onClick={[Function]}>
      x
    </div>
  </div>
</div>
`;

exports[`test WebMMediaElement DOM should render correctly 3`] = `
<div>
  <img
    className="thumbnail-media__thumbnail thumbnail-media__thumbnail--show-raw"
    onClick={[Function]}
    src="thumbnailSrc" />
  <div
    className="thumbnail-media__raw-media-container">
    <video
      autoPlay={true}
      className="thumbnail-media__raw-media--webm"
      controls={true}
      onCanPlay={[Function]}
      src="rawMediaSrc" />
    <div
      className="thumbnail-media__raw-media-close-button"
      onClick={[Function]}>
      x
    </div>
  </div>
</div>
`;

exports[`test WebMMediaElement DOM should render correctly 4`] = `
<div>
  <img
    className="thumbnail-media__thumbnail"
    onClick={[Function]}
    src="thumbnailSrc" />
</div>
`;
